stout/event-loop.h

```diff
@@ -177,6 +177,51 @@ class EventLoop : public Scheduler {
     uv_signal_t* handle_ = nullptr;
   };
 
+  class Timer {
+   public:
+    Timer(EventLoop& loop) {
+      handle_ = new uv_timer_t();
+      CHECK_EQ(
+          uv_timer_init(loop, handle_),
+          0);
+    }
+
+    Timer(const Timer& that) = delete;
+
+    Timer(Timer&& that) {
+      handle_ = that.handle_;
+      that.handle_ = nullptr;
+    }
+
+    ~Timer() {
+      if (handle_) {
```

Comment:
As on the other PR, I'd love for us to do type-safe checks:
```suggestion
      if (handle_ != nullptr) {
```